From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: The Batocera Team <batocera-linux@users.noreply.github.com>
Date: Thu, 1 Jan 2026 00:00:00 +0000
Subject: [PATCH] Keep a single default hotkey

toggle_hud is Shift_R+Undo by default and every other hotkey is unbound
by default. They can still be set in MangoHud.conf.

Taken from batocera.linux, package/batocera/utils/mangohud/005-hotkeys.patch
(LGPL-3.0-or-later, Copyright The Batocera Team).

diff --git a/src/overlay_params.cpp b/src/overlay_params.cpp
index b10a0cea5890f4f12591d9877da5992e2b30ec07..0be38dc3ea0e6de9acd66baa9e9f50c7ae2d7ccb 100644
--- a/src/overlay_params.cpp
+++ b/src/overlay_params.cpp
@@ -973,15 +973,15 @@ parse_overlay_config(struct overlay_params *params,
    }
 
 #if defined(HAVE_X11) || defined(HAVE_WAYLAND)
-   params->toggle_hud = { XKB_KEY_Shift_R, XKB_KEY_F12 };
-   params->toggle_hud_position = { XKB_KEY_Shift_R, XKB_KEY_F11 };
-   params->toggle_preset = { XKB_KEY_Shift_R, XKB_KEY_F10 };
-   params->reset_fps_metrics = { XKB_KEY_Shift_R, XKB_KEY_F9};
-   params->toggle_fps_limit = { XKB_KEY_Shift_L, XKB_KEY_F1 };
-   params->toggle_logging = { XKB_KEY_Shift_L, XKB_KEY_F2 };
-   params->reload_cfg = { XKB_KEY_Shift_L, XKB_KEY_F4 };
-   params->upload_log = { XKB_KEY_Shift_L, XKB_KEY_F3 };
-   params->upload_logs = { XKB_KEY_Control_L, XKB_KEY_F3 };
+   params->toggle_hud = { XKB_KEY_Shift_R, XKB_KEY_Undo };
+   params->toggle_hud_position = { };
+   params->toggle_preset = { };
+   params->reset_fps_metrics = { };
+   params->toggle_fps_limit = { };
+   params->toggle_logging = { };
+   params->reload_cfg = { };
+   params->upload_log = { };
+   params->upload_logs = { };
 #endif
 
 #ifdef _WIN32
